packages/x-date-pickers/src/MonthCalendar/PickersMonth.tsx

```diff
@@ -47,12 +47,17 @@ const PickersMonthRoot = styled('div', {
   overridesResolver: (_, styles) => [styles.root],
 })<{
   ownerState: PickersMonthProps;
-}>(({ ownerState }) => ({
-  flexBasis: ownerState.monthsPerRow === 3 ? '33.3%' : '25%',
+}>({
   display: 'flex',
   alignItems: 'center',
   justifyContent: 'center',
-}));
+  variants: [3, 4].map((monthsPerRow) => ({
+    props: { monthsPerRow },
+    style: {
+      flexBasis: `${Math.round((1 / monthsPerRow) * 1000) / 10}%`,
+    },
+  })),
```

Comment:
Based on the previous logic, if `monthsPerRow = 1` it would not work.

I would make it explicit like this:

```js
variants: [
  { props: { monthsPerRow: 3 }, style: { flexBasis: '33%' } }
  { props: ({ ownerState }) => ownerState.monthsPerRow !== 3, style: { flexBasis: '25%' } }
]
```

Reply:
It cannot be equal to 1, the props has `3 | 4` as a type, is there any benefit to support values that we don't intend to support?

Reply:
@siriwatknp

Reply:
Got it then I'd recommend not using `.map` and split them out like this:

```js
variants: [
  { props: { monthsPerRow: 3 }, style: { flexBasis: '33%' } },
  { props: { monthsPerRow: 4 }, style: { flexBasis: '25%' } }
]
```

I think `.map` should only be used to handle extension from developers, e.g. `theme.palette`:

```js
variants: [
  …(Object.entries(theme.palette)).map(…)
]
```